### PR TITLE
Adding sections for memory tweaking and docker stats

### PR DESCRIPTION
**Pull request by danielbrutti**
Hi,

I just added information about a docker stats command and a new section for memory tweaking in case someone else had the [issue](https://github.com/jhipster/generator-jhipster/issues/3791) I had.

Thanks,

Daniel

### CODE REVIEW

**Comment by danielbrutti** [comment]
Hi.. I found something.. setting just the environment section on docker-compose.yml doesn't work for me

- JAVA_OPTS="-Xmx512m -Xmx256m"

Please, do not approve this doc changes yet.. will continue working on this.

**Comment by pascalgrimaud** [comment]
Probably a problem in syntax, with double quote

**Comment by danielbrutti** [comment]
Is working, but I need to add `${JAVA_OPTS}` into Dockerfile CMD line on each project to make sure the opts are added to java -jar command.

I can add this into the docs.. but I'm wondering if there is another way to make sure it uses JAVA_OPTS without modify Dockerfile.

**Comment by danielbrutti** [comment]
Also.. the problem arises when executing the registry, because the memory parameters are not used when registry is in docker compose:
~~~
jhipster-registry:
        extends:
            file: jhipster-registry.yml
            service: jhipster-registry
        environment:
            - JAVA_OPTS="-Xmx512m -Xmx256m"
~~~

and jhipster-registry.yml defines the image from docker hub.

~~~
image: jhipster/jhipster-registry:v2.2.0
~~~

In this case, I don't have control on CMD line to setup ${JAVA_OPTS}

**Comment by danielbrutti** [comment]
OK found it.. it works when setting up this "_JAVA_OPTIONS" environment variable... this one is used by default by the JVM. If you agree, I can modify docs using this variable instead of JAVA_OPTS

**Comment by PierreBesson** [comment]
@danielbruti It's strange, I thought _JAVA_OPTIONS was the old name of the variable. Also it shouldn't be necessary to modify the Dockerfile to edit the env var from docker-compose. But if it happens to be the case, I will add this to the jhipster's app and registry's Dockerfile on the CMD line.
 I will check everything tomorrow and come back to you.
Cc @pascalgrimmaud

**Comment by danielbrutti** [comment]
@PierreBesson setting up `_JAVA_OPTIONS` in docker-compose.yml is not necessary to add the variable in Dockerfile. CMD can remain as it is right now.

**Comment by PierreBesson** [comment]
Ok. I guess we can go with _JAVA_OPTIONS.

File: pages/docker_compose.md [hunk 1]
```diff
@@ -199,3 +214,28 @@ When you stop a container, the data is not deleted, unless you delete the contai
 Be careful! All data will be deleted:
 
 `docker rm <container_id>`
+
+
+## <a name="8"></a> Memory Tweaking
+
+In order to optimize memory usage for applications running in the container, you can setup Java memory parameters on Dockerfile or docker-compose.yml
```
Comment (jdubois) [inline pages/docker_compose.md]: Can you use backquotes around filenames? Like `docker-compose.yml`

**Comment by pascalgrimaud** [comment]
strange! I always use JAVA_OPTS

**Comment by pascalgrimaud** [comment]
@danielbrutti can you try exactly (without -> " ), and no additional "command" in docker-compose file
```
        environment:
            - JAVA_OPTS=-Xmx512m -Xmx256m
```

**Comment by jdubois** [comment]
Is this ready to be merged or not? Doesn't look like you agree on the comments... And I also did one comment in the code that wasn't closed.

**Comment by danielbrutti** [comment]
Hi, I'll finish this today, had hard weeks on work. 
Sorry for the delay.

**Comment by danielbrutti** [comment]
Just pushed last changes. It did not work with 
~~~
- JAVA_OPTS=-Xmx512m -Xmx256m
~~~
So I left `_JAVA_OPTIONS` in docs too.

File: pages/docker_compose.md (hunk 1 above)
Comment (danielbrutti) [inline pages/docker_compose.md]: Done.. I also modified raw data. `docker stats {% raw %}$(docker ps --format={{.Names}}){% endraw %}`

**Comment by jdubois** [comment]
I didn't know about this `_JAVA_OPTIONS` setting, and found some info on [this StackOverflow question](http://stackoverflow.com/questions/28327620/difference-between-java-options-java-tool-options-and-java-opts).
I'm not a big fan of it, it's an undocumented option, and `JAVA_OPTS` should work everywhere -> I would like to remove the part on this

**Comment by danielbrutti** [comment]
OK, I will remove it.. I'm not a fan of use anything different of JAVA_OPTS... but for some reason it didn't work for me. However, I'm ok with remove _JAVA_OPTIONS since is not the standard.

**Comment by pascalgrimaud** [comment]
@danielbrutti : for your information, with my current customer, we manage to run a full stack microservices with Docker on our test environment. I didn't put any specific configuration about memory, everything works perfectly !

**Comment by danielbrutti** [comment]
@pascalgrimaud yes, I read is working for everyone, just failing to me :P 
That's why I'm ok with remove that option from docs.. because is more probably to be an issue from my side or OS or anything else, completely different to JHipster + Docker.

**Comment by danielbrutti** [comment]
I just removed the options from docs.

**Comment by pascalgrimaud** [comment]
@danielbrutti : your analyze helps me a lot on my current project, we had to optimize our memory on our machines. So big thanks!
You were totally right about JAVA_OPTS. It didn't work on docker-compose, but _JAVA_OPTIONS works!

I will analyze and do more tests, then I will PR an update on this documentation.

**Comment by danielbrutti** [comment]
@pascalgrimaud : glad to hear that it did helped you!
We are still using _JAVA_OPTIONS, I really don't know why it does not work with JAVA_OPTS.. indeed we must use it as parameter when starting the registry with docker run too:
```
docker run -e "_JAVA_OPTIONS=-Xmx512m -Xmx256m" -p=8761:8761 -it jhipster/jhipster-registry:v2.2.0
```
Let me know if there is anything I can help.

**Comment by pascalgrimaud** [comment]
I can see Xmx, 2 times. Maybe you miss Xms ?

**Comment by danielbrutti** [comment]
@pascalgrimaud yes you are right, is Xms. Thanks!

**Comment by jgasmi** [comment]
The JAVA_OPTS works fine for me as shown below in case this may help

`environment:
            - "JAVA_OPTS=-Xmx512m -Xmx256m"`

**Pull request merged**